### ISSUE TITLE
Attempt to execute code removed by Dart AOT compiler (TFA) in 3.0.0-dev.1

### ISSUE THREAD

**Issue by gruuna** [cbl-dart/cbl-dart#545]
**Describe the bug**

In our app we switched already to development version 3.0.0-dev.1 because of some other dependency issues... 
When starting a replication (creating a AsyncReplicator) cbl raises the following error when in RELEASE mode (AOT):

**Attempt to execute code removed by Dart AOT compiler (TFA)**

Stacktrace:

  Channel.call.<anonymous closure> (package:cbl/src/service/channel.dart:156)
  ProxyReplicator.create (package:cbl/src/replication/proxy_replicator.dart:98)

**To Reproduce**

1. used ReplicatorConfiguration:

`final ReplicatorConfiguration replicatorConfiguration = ReplicatorConfiguration(
        target: UrlEndpoint(Uri.parse('ws://localhost:4984/db/')),
        authenticator: BasicAuthenticator(username: 'foo', password: 'bar'),
      );
      final CollectionConfiguration collectionConfiguration =
          CollectionConfiguration(
              channels: ['common']);
      replicatorConfiguration.addCollection(
          collection, collectionConfiguration);`

2. create Replicator with the config:

`final AsyncReplicator replicator =
          await Replicator.createAsync(replicatorConfiguration);`

**Environment:**

 - OS: macOS & iOS
 - `flutter`: 3.16.4
 - `cbl`: 3.0.0-dev.1
 - `cbl_flutter`: 3.0.0-dev.1
 - `cbl_flutter_ee` or `cbl_flutter_ce`: 3.0.0-dev.1

**Comment by blaugold** [comment]
@gruuna Thanks for reporting this! I can reproduce the issue, and it looks like a bug in the Dart SDK. I've filed in issue over there: https://github.com/dart-lang/sdk/issues/54414

**Comment by gruuna** [comment]
Maybe some information for fixing in here: https://github.com/flutter/flutter/issues/137297 or https://github.com/flutter/flutter/issues/51828 ?

**Comment by blaugold** [comment]
> Maybe some information for fixing in here: [flutter/flutter#137297](https://github.com/flutter/flutter/issues/137297) or [flutter/flutter#51828](https://github.com/flutter/flutter/issues/51828) ?

Maybe, but those particular issues seem to have been fixed a while ago and my guess is this is a new bug.

**Comment by gruuna** [comment]
Sorry for my spam :-)

It seems that dart removes the usage of the tuple in ffi_replicator.dart (Line 60):

`final (database, collections) = await resolveReplicatorCollections<SyncDatabase, SyncCollection, FfiDatabase, FfiCollection>(config);`

**Comment by blaugold** [comment]
> Sorry for my spam :-)

No problem :)

> It seems that dart removes the usage of the tuple in ffi_replicator.dart (Line 60):
> 
> `final (database, collections) = await resolveReplicatorCollections<SyncDatabase, SyncCollection, FfiDatabase, FfiCollection>(config);`

What makes you think that? I didn't try to dig deeper than replicating the exception because the last stack frame only contains the function name and no line numbers:

```
Unhandled exception:
Attempt to execute code removed by Dart AOT compiler (TFA)
#0      FfiReplicator.create (package:cbl/src/replication/ffi_replicator.dart)
#1      SyncReplicator.create (package:cbl/src/replication/replicator.dart:278)
#2      Replicator.createSync (package:cbl/src/replication/replicator.dart:147)
#3      main (file:///Users/terwesten/dev/repo_cbl_code_removed_tfa/main.dart:14)
<asynchronous suspension>
```

**Comment by gruuna** [comment]
I simply inserted print()'s at several locations in the create method :-) and so it pointed out that the mentioned line was the 'problematic' one...

Then I have rewritten the line to this (stupid, I know...):

```dart
FfiDatabase? database;
Map<FfiCollection, CollectionConfiguration>? collections;
await resolveReplicatorCollections<
    SyncDatabase, SyncCollection, FfiDatabase, FfiCollection>(config)
    .then((value) {
  database = value.$1;
  collections = value.$2;
});
``` 

And the errors from your reproduction-repo and also my error using the AsyncReplicator are gone...
So I think that dart removes this tuple-declaration...

**Comment by blaugold** [comment]
Thanks, @gruuna! That helped me create a minimal reproducible example in the issue in the Dart SDK repository. I can also create a workaround with that information.

**Comment by blaugold** [comment]
This should be fixed by https://github.com/cbl-dart/cbl-dart/commit/a68456e95d970c9e9344f73b7c88815233750dfe which is available in `cbl` `3.0.0-dev.2`.
